### PR TITLE
Bug fix to correct difference tabulation 

### PR DESCRIPTION
**Pull request by opotowsky**
## Description

🐞 alert in `_compareSets`!

The length of a set difference was wrong when src has parameters not in ref. Easy little fix! @Nebbychadnezzar our discussion led to this bug find!

`test_compareDatabaseSim` failed after the bug fix because it relied on the previously faulty `len(src - ref)` check returning 0. Because the test added extra time steps to the database, the bug fix now results in `compareDatabases` returning an `nDiff` of `None`. I simplified the test to insert a single parameter difference. `nDiffs()` is 3 because the `mean(x)`, `mean(abs(x))`, and `max(abs(x))` are all counted.

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Comment by jakehader** [comment]
I understand the fix but don't necessarily understand the optional name without looking at how this function is used in the codebase. This error message isn't really helpful as is and if I was to get this as a user I would not know what to do. 

Let me dig around some more but maybe @john-science can offer up an idea when he's got some time.

**Comment by opotowsky** [comment]
> > Edit: I don't see anywhere in the wild where the optional name argument is not provided. This can probably be required? What was the case where you saw a None?
> 
> You're correct -- I think anywhere that uses this in the code provides that argument. I did not provide the `name` arg when I wrote the test initially, which is how I got the above failure. I'm fine making it required, since it doesn't break any other code.

Instead of making it required I fixed the string processing. Wasn't feeling strongly one way or the other

**Comment by opotowsky** [comment]
Seems that `test_compareDatabaseSim` relied on this bug. Fixing.

File: armi/bookkeeping/db/compareDB3.py [hunk 1]
```diff
@@ -493,8 +490,10 @@ def _compareComponentData(
 
         if srcSpecial ^ refSpecial:
             out.writeln(
-                "Could not compare data because one uses special formatting, "
-                "and the other does not. Ref: {} Src: {}".format(refSpecial, srcSpecial)
+                "Could not compare data for parameter {} because one uses special "
+                "formatting, and the other does not. Ref: {} Src: {}".format(
+                    paramName, refSpecial, srcSpecial
+                )
```
Comment (opotowsky) [inline armi/bookkeeping/db/compareDB3.py:496]: This was a tiny update I made and forgot to PR. Since it's in the same file, figured I could sneak it in 😉

**Review by jakehader — APPROVED** [review]

**Pull request merged**